src/Proto.Cluster/Partition/PartitionIdentityRelocationWorker.cs

```
@@ -0,0 +1,176 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Proto.Cluster.Partition
+{
+    /// <summary>
+    /// Used by partitionIdentityActor to update partition lookup on topology changes.
+    /// </summary>
+    class PartitionIdentityRelocationWorker : IActor
+    {
+        private static readonly ILogger Logger = Log.CreateLogger<PartitionIdentityActor>();
+
+        private readonly Dictionary<ClusterIdentity, PID> _partitionLookup;
+        private readonly TaskCompletionSource<bool> _onRelocationComplete = new();
+        private readonly Dictionary<PID, MemberRequestState> _waitingRequests = new();
+        private int _totalReceived;
+        private readonly Stopwatch _timer = new();
+
+        public PartitionIdentityRelocationWorker(Dictionary<ClusterIdentity, PID> partitionLookup) => _partitionLookup = partitionLookup;
+
+        public Task ReceiveAsync(IContext context) => context.Message switch
+        {
+            IdentityHandoverRequest request   => OnIdentityHandoverRequest(request, context),
+            IdentityHandoverResponse response => OnIdentityHandoverResponse(response, context),
+            DeadLetterResponse response       => OnDeadLetterResponse(response, context),
+            ReceiveTimeout                    => OnReceiveTimeout(),
+            _                                 => Task.CompletedTask
+        };
+
+        private Task OnReceiveTimeout()
+        {
+            // TODO: Retry strategy
+            Logger.LogError("Relocation timed out");
+            _onRelocationComplete.TrySetResult(false);
+            return Task.CompletedTask;
+        }
+
+        private Task OnDeadLetterResponse(DeadLetterResponse response, IContext context)
+        {
+            if (_waitingRequests.Remove(response.Target))
+            {
+                Logger.LogError("Unreachable node: {Address}", response.Target.Address);
```

How is the best way to handle network partitions between members? If this is from a member which should still be in the cluster, we would be in an inconsistent state.

---

More likely than network partition is that the member has already shut down though, which means that the node will need to rebalance again, and we will end up in a correct state.

---

Any member that become invisible to other members due to network partitioning, would be evicted from the cluster.
We have a fairly different approach than e.g. Akka which tries to "fix" things.
But in the era of let it crash, containers and pods.
Just let that node shut down, and restart